packages/api/portfolioDrafts/portfolio/createPortfolioStep.ts

```diff
@@ -1,90 +1,76 @@
-// import { createPortfolioStepCommand } from "../../utils/commands/createPortfolioStepCommand";
-import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
-import { DynamoDBDocumentClient, UpdateCommand, UpdateCommandOutput } from "@aws-sdk/lib-dynamodb";
-import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
 import { PORTFOLIO_STEP } from "../../models/PortfolioDraft";
 import { PortfolioStep } from "../../models/PortfolioStep";
-import { ApiSuccessResponse, ErrorStatusCode, OtherErrorResponse, SuccessStatusCode } from "../../utils/response";
-import { isBodyPresent, isPathParameterPresent, isPortfolioStep, isValidJson } from "../../utils/validation";
-
-const TABLE_NAME = process.env.ATAT_TABLE_NAME ?? "";
-export const NO_SUCH_PORTFOLIO = new OtherErrorResponse(
-  "Portfolio Draft with the given ID does not exist",
-  ErrorStatusCode.NOT_FOUND
-);
-export const REQUEST_BODY_INVALID = new OtherErrorResponse(
-  "A valid PortfolioStep object must be provided",
-  ErrorStatusCode.BAD_REQUEST
-);
-export const EMPTY_REQUEST_BODY = new OtherErrorResponse("Request body must not be empty", ErrorStatusCode.BAD_REQUEST);
+import { dynamodbDocumentClient as client } from "../../utils/dynamodb";
+import { NO_SUCH_PORTFOLIO_DRAFT } from "../../utils/errors";
+import { ApiSuccessResponse, SuccessStatusCode, SetupError, DatabaseError, DatabaseResult } from "../../utils/response";
+import { postRequestShapeValidation } from "../../utils/requestValidation";
 
 /**
  * Submits the Portfolio Step of the Portfolio Draft Wizard
  *
  * @param event - The POST request from API Gateway
  */
-export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
-  if (!isBodyPresent(event.body)) {
-    return EMPTY_REQUEST_BODY;
-  }
-
-  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
-
-  if (!isPathParameterPresent(portfolioDraftId)) {
-    return NO_SUCH_PORTFOLIO;
-  }
 
-  if (!isValidJson(event.body)) {
-    return REQUEST_BODY_INVALID;
+export async function handler(event: APIGatewayProxyEvent, context?: Context): Promise<APIGatewayProxyResult> {
+  // Perform shape validation
+  const setupResult = postRequestShapeValidation<PortfolioStep>(event);
```

Comment:
`post` feels like it implies something here and keeps confusing me. like "shape validation after the request". Can we do something like `shapeValidationForPostRequest`?